fab/fabfile.py

```diff
@@ -107,66 +110,62 @@ def _require_target():
 
 
 class DeployMetadata(object):
-    def __init__(self, env):
-        self.env = env
+    def __init__(self, code_branch, environment):
         self.timestamp = datetime.datetime.utcnow().strftime('%Y-%m-%d_%H.%M')
         self._deploy_ref = None
         self._deploy_tag = None
-        self._git = sh.git.bake(_tty_out=False)
+        self._github = _get_github()
+        self._repo = self._github.repository('dimagi', 'commcare-hq')
+        self._max_tags = 100
+        self._last_tag = None
+        self._code_branch = code_branch
+        self._environment = environment
 
     def tag_commit(self):
-        self._git.fetch("origin", "--tags")
-        pattern = "*{}*".format(self.env.environment)
-        self.last_tag = sh.tail(self._git.tag("-l", pattern), "-1").strip()
-
-        tag_name = "{}-{}-deploy".format(self.timestamp, self.env.environment)
-        msg = "{} deploy at {}".format(self.env.environment, self.timestamp)
-        self._git.tag(tag_name, "-m", msg, self.deploy_ref)
-        self._git.push("origin", tag_name)
+        pattern = ".*-{}-.*".format(self._environment)
+        for tag in self._repo.tags(self._max_tags):
+            if re.match(pattern, tag.name):
+                self._last_tag = tag.name
+                break
+
+        if not self._last_tag:
+            print yellow('Warning: No previous tag found in last {} tags for {}'.format(
+                self._max_tags,
+                self._environment
+            ))
+        tag_name = "{}-{}-deploy".format(self.timestamp, self._environment)
+        msg = "{} deploy at {}".format(self._environment, self.timestamp)
+        user = self._github.me()
+        self._repo.create_tag(
+            tag=tag_name,
+            message=msg,
+            sha=self.deploy_ref,
+            obj_type='commit',
+            tagger={
+                'name': user.name,
+                'email': user.email,
+            }
+        )
         self._deploy_tag = tag_name
```

Comment:
I think this isn't actually creating tags.  I tried this:
```python
In [1]: from fab.fabfile import *
code_branch not specified, using 'master'. You can set it with '--set code_branch=<branch>'

In [2]: md = DeployMetadata('testing', 'staging')

In [3]: md.tag_commit()
```
and couldn't find a tag either locally or on github.  It looks like there have been [no new tags](https://github.com/dimagi/commcare-hq/tags) for the last two days, so I think the recent deploys haven't been tagged.